File: 30_05_2024_Count_Triplets_That_Can_Form_Two_ Arrays_of_Equal_XOR.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int ans = 0;
        for (int i = 0; i < arr.size(); i++) {
            int xora = 0;
            for (int j = i + 1; j < arr.size(); j++) {
                xora ^= arr[j - 1];
                int xorb = 0;
                for (int k = j; k < arr.size(); k++) {
                    xorb ^= arr[k];
                    if (xora == xorb) {
                        ans++;
                    }
                }
            }
        }
        return ans;
    }
};

File: 30_05_2024_Count_Triplets_That_Can_Form_Two_ Arrays_of_Equal_XOR.py
class Solution(object):
    def countTriplets(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        ans=0
        for i in range(len(arr)):
            xorA=0
            for j in range(i+1,len(arr)):
                xorA=xorA^arr[j-1]
                xorB=0
                for k in range(j,len(arr)):
                    xorB=xorB^arr[k]
                    if xorA==xorB:
                        ans+=1
        return ans
